### PR TITLE
Doc: Add CodeCov report to travis CI

### LINKED ISSUE #171: Dev: Add CodeCov Report to Travis CI
## Description
As a maintainer,
I need to add CodeCov report to the Travis CI,
so that I can make sure the code quality written in the PRs.

## Mocks
NA

### Update [Required]
- [ ] Update Travis CI to add CodeCov report

### Enhancement to Update [Optional]
NA

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 hr

### PR DESCRIPTION
**Pull request by mtreacy002**
### Description
Added CodeCov report to Travis CI

Fixes #171 

### Type of Change:
- Code
- Quality Assurance

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Pushing this PR to develop showed Travis CI build as pass and generated CodeCov report:
* on CodeCov badge
Screen Shot 2020-10-26 at 12 11 58 am

* on CodeCov.io
Screen Shot 2020-10-26 at 12 19 41 am

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by mtreacy002** [comment]
Hi @devkapilbansal , I accidentally reset the commits back on the original PR #171 🤣🤣. Can you please provide your review here? Thanks beforehand.

File: .travis.yml [hunk 1]
```diff
@@ -28,3 +28,5 @@ install:
 script:
   - python -m unittest discover tests -v
   - pytest --cov tests
+  - bash <(curl -s https://codecov.io/bash)
+
```
Comment (devkapilbansal) [inline .travis.yml:32]:
```suggestion

```

File: .travis.yml [hunk 2]
```diff
@@ -28,3 +28,5 @@ install:
 script:
   - python -m unittest discover tests -v
```
Comment (devkapilbansal) [inline .travis.yml:29]: Why `coverage` is not used instead of `python`. Similar to what is done in #172

**Comment by devkapilbansal** [comment]
> Hi @devkapilbansal , I accidentally reset the commits back on the original PR #171 . Can you please provide your review here? Thanks beforehand.

Hi @mtreacy002 if pull request get accidentally closed because of 0 commits, then you can reopen it after pushing to the branch. The only condition is branch is not deleted from remote and pull request is not closed by maintainers(in case of outside collaborator)

File: .travis.yml (hunk 1 above)
Comment (Rahulm2310) [reply .travis.yml:32]: @devkapilbansal don't you suggest terminating with a new line ?

Comment (devkapilbansal) [reply .travis.yml:32]: I do but that line contains spaces. And blank line shouldn't contain anything

**Comment by mtreacy002** [comment]
> Hi @mtreacy002 if pull request get accidentally closed because of 0 commits, then you can reopen it after pushing to the branch. The only condition is branch is not deleted from remote and pull request is not closed by maintainers(in case of outside collaborator)

Ok, will do that next time. I was panicking coz I accidentally did the reset --hard when I tried to squash my commits 🤣🤣.

File: .travis.yml (hunk 2 above)
Comment (mtreacy002) [reply .travis.yml:29]: I'm not sure I'm folllowing, @devkapilbansal . Isn't the next line (line 30) means we're using coverage?

File: .travis.yml (hunk 1 above)
Comment (mtreacy002) [reply .travis.yml:32]: Done 😉

**Comment by Rahulm2310** [comment]
@mtreacy002 just a thought. Aren't we moving to `github-actions` from `travis`. If yes, wouldn't it be better to add `codecov` to `github-actions`.

**Comment by mtreacy002** [comment]
@Rahulm2310 , we are, but until the GitHub action PR is approved, Travis will still be used. Both Travis and GitHub action need to have the coverage.
PS: The initial PR was opened 25th Oct 2020 but unfortunately only got noticed and reviewed this month 😔.

**Comment by mtreacy002** [comment]
closing this issue now as we're moving to github action

**Pull request closed without merging**